### DISCUSSION TITLE
LinkedIn as OIDC provider

### DISCUSSION THREAD

**Discussion by alan994** [ory/kratos#2497] [Q&A]
Hi,

Is there an example of how to use `LinkedIn` as an OIDC provider? I tried to use a generic provider and pass it `authorization_endpoint` and `token_endpoint` but, I got an error saying `Ory KRatos` can't fetch `discovery document` (JSON typically exposed on `/.well-known/openid-configuration). Is there any way I can tell `Kratos` to use only `auth` and `token` endpoints instead of trying to read it from the discovery endpoint? 

I saw some PR regarding `LinkedIn` integration but not sure when it will be merged. If someone managed to get LinkedIn working with `Ory Kratos` please share your configuration.

**Comment by vinckr** [comment]
Hello Alan,
you could already use the code in https://github.com/ory/kratos/pull/2207 - the 

LinkedIn does not actually implement OIDC, so there is some workaround needed:

>LinkedIn like Github does not implement OpenID connect and just like the github provider, this new linkedin provider will call the linkedin API to retrieve information for claims.

**Reply by ramirobg94** [reply]
other here

**Comment by ncarkaci** [comment]
You can access the Linkedin OpenID configuration page here. https://www.linkedin.com/oauth/.well-known/openid-configuration

**Reply by duartegarin** [reply]
So it does implement OIDC or not? We're having the exact situation and unfortunately missed this in the scoping Ory stage.
